session/schema_amender.go

```diff
@@ -319,7 +315,8 @@ type amendOperationDeleteOldIndex struct {
 
 // amendOperationAddNewIndex represents the add operation will be performed on new key values for add index amend.
 type amendOperationAddNewIndex struct {
-	info *amendOperationAddIndexInfo
+	info                  *amendOperationAddIndexInfo
+	processedNewIndexKeys map[string]interface{}
```

Comment:
If we use `struct{}` as the value type, it would be more clear that we only need the key, the value is not used.